Rotation matrix correction

LonglonWu opened the pull request.

LonglonWu committed:
Rotation matrix correction

mpmdean commented:
Thanks @LonglonWu 

While I agree that the prior version was wrong, I think this revised version is also wrong.

The `xcenter` and `ycenter` values are effectively doubled, so that the peak appears at 4 when
specified to be centered at 2.


fig, ax = plt.subplots()

X, Y = np.meshgrid(np.linspace(-10, 10), np.linspace(-10, 10))


I = lorentzianSq2DRot(X, Y, centerx=2, centery=3, sigmax=2, angle=0)

art = ax.pcolor(X, Y, I)
plt.colorbar(art, ax=ax)
ax.set_xlabel('x')
ax.set_ylabel('y')
ax.set_aspect('equal')

![image](https://user-images.githubusercontent.com/699991/95752999-ec942600-0c6e-11eb-942d-32fe45077cde.png)

LonglonWu committed:
For the 2D Lorentzian square,
the "centerx" and "centery" are forget to remove. Corrected.

mpmdean commented:
`lorentzianSq2DRot` looks correct, but `lorentzianSq3D` is wrong. What specifically did you think was wrong with this before?

LonglonWu committed:
Hi Mark,

For the 3D peak, the square is missed. I have added it into the scaript.
Also another lorentzian squared peak shape is added, which allow the peak rotate in x-y plane

mpmdean commented:
Thanks @LonglonWu
